src/CartesianLocalCoordinateSystem.C

```diff
@@ -0,0 +1,300 @@
+    mooseError(
+        "The local coordinate system must be provided by one of the following options: "
+        "(1) 'dip_direction_degree', 'dip_angle_degree' and 'dip_option'), "
+        "or (2) via input of trend and plunge of e1 and e2 ('e1_trend_degree', 'e1_plunge_degree', "
+        "'e2_trend_degree' and 'e2_plunge_degree'). "
+        "or (3) via input of vector and point ('e1' and 'e2' or 'point_on_e1_e2').");
+
+  if (from_dip > 0)
+  {
+    if (from_dip != 3)
+      mooseError("All parameters 'dip_direction_degree', 'dip_angle_degree' and 'dip_option' must "
+                 "be specified.");
+    buildFromDip(parameters);
+  }
+  else if (from_trend_and_plunge > 0)
+  {
+    if (from_trend_and_plunge != 4)
+      mooseError("Parameters 'e1_trend_degree', 'e1_plunge_degree', 'e2_trend_degree' and "
+                 "'e2_plunge_degree' must be specified.");
+    buildFromTrendAndPlunge(parameters);
+  }
+  else if (from_vectors > 0)
+  {
+    if (from_vectors != 2 || !parameters.isParamSetByUser("e1"))
+      mooseError(
+          "Parameter 'e1' must be given and one of the parameters 'e2' or 'point_on_e1_e2'.");
+    buildFromVectorAndPoint(parameters);
+  }
+  else
+    mooseError("Unknown data source. Are you missing a parameter? Did you misspell one?");
+
+  // build rotation matrix global to local
+  _rotate_global_to_local(0, 0) = _e1(0);
+  _rotate_global_to_local(1, 0) = _e1(1);
+  _rotate_global_to_local(2, 0) = _e1(2);
+  _rotate_global_to_local(0, 1) = _e2(0);
+  _rotate_global_to_local(1, 1) = _e2(1);
+  _rotate_global_to_local(2, 1) = _e2(2);
+  _rotate_global_to_local(0, 2) = _e3(0);
+  _rotate_global_to_local(1, 2) = _e3(1);
+  _rotate_global_to_local(2, 2) = _e3(2);
+
+  _rotate_local_to_global = _rotate_global_to_local.inverse();
+
+  // std::cout << "  " << name() << ".e1:\n" << _e1 << "\n";
+  // std::cout << "  " << name() << ".e2:\n" << _e2 << "\n";
+  // std::cout << "  " << name() << ".e3:\n" << _e3 << "\n";
+}
+
+void
+CartesianLocalCoordinateSystem::buildFromDip(const InputParameters & parameters)
+{
+  const auto dip_direction = parameters.get<Real>("dip_direction_degree") * (libMesh::pi / 180.0);
+  const auto dip_angle = parameters.get<Real>("dip_angle_degree") * (libMesh::pi / 180.0);
+  const std::string dip_option = parameters.get<MooseEnum>("dip_option");
+
+  // std::cout << "  buildFromDip\n";
+  // std::cout << "  dip_direction_degree = " << parameters.get<Real>("dip_direction_degree") << ")\n";
+  // std::cout << "  dip_angle_degree = " << parameters.get<Real>("dip_angle_degree") << ")\n";
+  // std::cout << "  dip_option = " << dip_option << ")\n";
+
+  const Real c1 = std::cos(dip_direction);
+  const Real s1 = std::sin(dip_direction);
+  const Real c2 = std::cos(dip_angle);
+  const Real s2 = std::sin(dip_angle);
+
+  if (dip_option == "e1_e2_plane_e1_horizontal")
+  {
+    // @Kavan-Khaledi: the formulas below mimic the 'old' behaviour of OpalinusElasticityTensor
```

Comment:
Yes this is correct, it should replicate the old behavior. Please note that in the opalinusElasticityTensor the dip was multiplied by (-1) because the first rotation in e1-e2-plane-e1-horizontal rule is counterclockwise

Reply:
Thanks for having a look! I double-checked the calculation and it should be fine now using ´dip_direction´ as is (otherwise one would had to invert ´dip_direction´ and ´s1´).